dot/network/transaction.go

```diff
@@ -119,36 +120,54 @@ func decodeTransactionHandshake(_ []byte) (Handshake, error) {
 	return &transactionHandshake{}, nil
 }
 
-func (s *Service) createBatchMessageHandler(txnBatch chan *BatchMessage) NotificationsMessageBatchHandler {
-	return func(peer peer.ID, msg NotificationsMessage) (msgs []*BatchMessage, err error) {
+func (s *Service) createBatchMessageHandler(txnBatchCh chan *BatchMessage) NotificationsMessageBatchHandler {
+	go func() {
+		protocolID := s.host.protocolID + transactionsID
+		ticker := time.NewTicker(s.cfg.SlotDuration)
+		defer ticker.Stop()
+
+		for {
+		out:
+			select {
+			case <-s.ctx.Done():
+				return
+			case <-ticker.C:
+				timeOut := time.NewTimer(s.cfg.SlotDuration / 3)
+				for {
+					select {
+					case <-timeOut.C:
+						break out
+					case txnMsg := <-txnBatchCh:
+						propagate, err := s.handleTransactionMessage(txnMsg.peer, txnMsg.msg)
```

Comment:
My question is: the `txnMsg` arrived but the following line takes more than a ticker duration
```go
propagate, err := s.handleTransactionMessage(txnMsg.peer, txnMsg.msg)
```
is that a problem?

Reply:
It won't be a problem as we need to process the extrinsic sequentially. We cannot spin a goroutine.

Reply:
The ticker duration won't be respected but there is not much solution there. Although, I'm not super familiar with the context, but my question:

if ticker duration `s.cfg.SlotDuration` is say `10` and processing takes `15`, are we fine executing the next processing at t+15 (instead of t+20)? Since the ticker will have its (1-buffered) channel blocked and will send its tick as soon as we finish our processing at `t+15`.

Reply:
Ideally, it will take very little time since we are just verifying transactions. But even if it does, it shouldn't matter since we are only adding transactions to the memory pool.